arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2976,7 +2992,18 @@ Result RestReplicationHandler::createBlockingTransaction(aql::QueryId id,
   TRI_ASSERT(isLockHeld(id).is(TRI_ERROR_HTTP_NOT_FOUND));
 
   try {
-    queryRegistry->insert(id, query.get(), ttl, true, true);
+    std::function<void(void)> f =
+      [=]() {
+        try {
+          // Code does not matter, read only access, so we can roll back.
+          queryRegistry->destroy(_vocbase.name(), id, TRI_ERROR_QUERY_KILLED, false);
+        } catch (...) {
+          // All errors that show up here can only be
+          // triggered if the query is destroyed in between.
+        }
+      };
+    auto rGuard = std::make_unique<CallbackGuard>(f);
+    queryRegistry->insert(id, query.get(), ttl, true, true, std::move(rGuard));
```

Comment:
You can also create a CallbackGuard and move it directly. Just fyi, the unique_ptr is fine, too.

Reply:
i chose the unique ptr to not have to move an empty CallbackGuard and for the personally perceived nicer check on nullptr in case none is handed over to `QueryRegistry::insert` and in `QueryInfo`

Reply:
👍